llm.el

```diff
@@ -128,6 +131,52 @@ REQUIRED is whether this is required or not."
   type
   required)
 
+(cl-defstruct llm-media
+  "Contains media that can be sent as part of an interaction.
+
+MIME-TYPE is a string containing the mime type of the media.
```

Comment:
This might be a good place to warn that not all mime types are accepted by all providers

Reply:
done